### PR TITLE
Fixes for unicode decoding and missing file closing

### PR DESCRIPTION
**Pull request by tkupek**

### CODE REVIEW

File: darwin/dataset/remote_dataset.py [hunk 1]
```diff
@@ -249,7 +253,7 @@ def pull(
 
         if release.latest:
             latest_dir = self.local_releases_path / "latest"
-            if latest_dir.exists():
+            if os.path.islink(latest_dir):
```
Comment (andreaazzini) [inline darwin/dataset/remote_dataset.py:256]: I think we can leverage the `pathlib` equivalent
```suggestion
            if latest_dir.is_symlink():
```
We can avoid importing `os` that way

File: setup.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import setuptools
 
-with open("README.md", "r") as f:
-    long_description = f.read()
+with open("README.md", "rb") as f:
+    long_description = f.read().decode('utf-8')
```
Comment (andreaazzini) [inline setup.py:4]: I would add a comment on top of this line, explaining why decoding is important here

File: darwin/dataset/download_manager.py [hunk 3]
```diff
@@ -59,7 +59,9 @@ def download_all_images_from_annotations(
 
     annotations_to_download_path = []
     for annotation_path in annotations_path.glob(f"*.{annotation_format}"):
-        annotation = json.load(annotation_path.open())
+        with annotation_path.open() as file:
+            annotation = json.load(file)
+            file.close()
```
Comment (andreaazzini) [inline darwin/dataset/download_manager.py:64]: As far as I'm aware, files are automatically closed when control leaves the `with` block, so these `file.close()` calls can be avoided

**Review by andreaazzini — CHANGES_REQUESTED** [review]
Summary:
- Closing files is redundant at the end of `with` blocks
- We can avoid importing `os` and use its `pathlib` counterparts
- It would be nice to add a comment on why decoding is needed for future reference

File: darwin/dataset/remote_dataset.py (hunk 1 above)
Comment (tkupek) [reply darwin/dataset/remote_dataset.py:256]: Pathlib `is_symlink()` breaks the code on my Windows 10 machine. Can you verify if it works for you?

File: setup.py (hunk 2 above)
Comment (tkupek) [reply setup.py:4]: @florian-fischer-swarm can you do that?

File: darwin/dataset/download_manager.py (hunk 3 above)
Comment (tkupek) [reply darwin/dataset/download_manager.py]: Done.

File: darwin/dataset/remote_dataset.py (hunk 1 above)
Comment (tkupek) [reply darwin/dataset/remote_dataset.py]: Done.

File: darwin/dataset/remote_dataset.py [hunk 4]
```diff
@@ -249,7 +250,7 @@ def pull(
 
         if release.latest:
             latest_dir = self.local_releases_path / "latest"
-            if latest_dir.exists():
+            if latest_dir.is_symlink():
```
Comment (andreaazzini) [inline darwin/dataset/remote_dataset.py:253]: This change seems to be working on macOS, but I don't have access to a Windows machine. Do you happen to have further information on why it breaks on Windows? That would be super helpful 🙂 

Also, is there a reason why we didn't like `latest_dir.exists()`?

File: setup.py (hunk 2 above)
Comment (andreaazzini) [reply setup.py:4]: I would also use double quotes `"`, so the linter does not complain 🙂 
```suggestion
    long_description = f.read().decode("utf-8")
```

**Review by andreaazzini — CHANGES_REQUESTED** [review]
Thank you for the quick updates!

The PR is looking pretty close to be mergeable, but I added a couple other comments.

File: darwin/dataset/remote_dataset.py (hunk 4 above)
Comment (tkupek) [reply darwin/dataset/remote_dataset.py:253]: `latest_dir.exists()` breaks on my Windows machine, `is_symlink` and the os alternative work.
I think `exists()` is meant for directories and files only, Windows handles symlinks a bit different as Unix systems, wherefore it does not work.

Comment (tkupek) [reply darwin/dataset/remote_dataset.py:253]: Maybe you can add some unit-tests and schedule them on different machines via GitHub actions.

File: setup.py [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import setuptools
 
 with open("README.md", "rb") as f:
-    long_description = f.read().decode('utf-8')
+long_description = f.read().decode("utf-8")
```
Comment (andreaazzini) [inline setup.py:4]: Indentation seems off here

**Review by andreaazzini — CHANGES_REQUESTED** [review]
One more thing 🙂 

There seems to be an indentation problem. After that's resolved, I'll approve and merge this PR.

Thank you!

**Review by andreaazzini — APPROVED** [review]

**Pull request merged**